pkgs/development/compilers/julia/patches/1.6/0005-nix-Enable-parallel-unit-tests-for-sandbox.patch

```diff
@@ -0,0 +1,30 @@
+From f35d62abfbd8f200e26c8081909bb4f61d866593 Mon Sep 17 00:00:00 2001
+From: Pontus Stenetorp <pontus@stenetorp.se>
+Date: Thu, 8 Apr 2021 04:37:44 +0000
+Subject: [PATCH 5/6] nix: Enable parallel unit tests for sandbox
+
+Disabled by default due to lack of networking in the Nix sandbox. This
+greatly speeds up the build process on a multi-core system.
+---
+ test/runtests.jl | 5 +++--
+ 1 file changed, 3 insertions(+), 2 deletions(-)
+
+diff --git a/test/runtests.jl b/test/runtests.jl
+index 2f9cd058bb..2f8c19fa32 100644
+--- a/test/runtests.jl
++++ b/test/runtests.jl
+@@ -83,8 +83,9 @@ prepend!(tests, linalg_tests)
+ import LinearAlgebra
+ cd(@__DIR__) do
+     n = 1
+-    if net_on
+-        n = min(Sys.CPU_THREADS, length(tests))
++    if net_on || haskey(ENV, "NIX_BUILD_CORES")
```

Comment:
Is this a “sane” way to detect the Nix build environment? I was not able to come up with something more clever.

Reply:
Wait, is the plan to officially accomodate upstream the special case relevant to Nix building? If the patch is Nixpkgs-only we could assume we are talking about a Nix build anyway…

Why parallel running of tests is off if net is not enabled but there are multiple local cores? Maybe there could be a generic override for the Nix expression to trigger?

Should single-core Nox build still use parallelism?

Reply:
I brought this up here: https://github.com/JuliaLang/julia/pull/40348 and eventually merged a comment to better explain the situation here: https://github.com/JuliaLang/julia/pull/40359. 

IIUC there's no reason not to just do `if true` when building with nix. The `net_on` test is more conservative than need be. It has something to do with julia supporting some build environments that disable access to the loopback interface as well as the broader network.

Reply:
I am not planning on upstreaming this as it only really “fails” inside our build sandbox. However, I am somewhat reluctant to make it `if true` as the patch will stick around post-install and I see no reason to fundamentally change how tests are run at that point.

Reply:
No, the patched source will be what is installed along with the derivation. So `if true` will lead to anyone running the test suite, even post-install, getting this behaviour.